src/amo/pages/Feedback/index.js

```diff
@@ -0,0 +1,478 @@
+        addonSlug: addonIdentifier,
+        reporter_email: email,
+        reporter_name: name,
+        message: text,
+        reason: category,
+      }),
+    );
+  };
+
+  getFormValues(currentUser: UserType | null): FormValues {
+    const defaultFormValues = {
+      email: '',
+      name: '',
+      text: '',
+      category: null,
+    };
+
+    const { email, display_name: name } = currentUser || {};
+
+    return {
+      ...defaultFormValues,
+      email: email || '',
+      name: name || '',
+    };
+  }
+
+  preventSubmit(): boolean {
+    const { loading } = this.props;
+    const { text, category } = this.state;
+
+    return loading || !category || !text || (text && text.trim() === '');
+  }
+
+  render(): React.Node {
+    const {
+      _config,
+      errorHandler,
+      i18n,
+      loading,
+      currentUser,
+      addonIdentifier,
+      abuseReport,
+    } = this.props;
+
+    if (!_config.get('enableFeatureFeedbackForm')) {
+      return <NotFoundPage />;
+    }
+
+    let errorMessage;
+
+    if (errorHandler.hasError()) {
+      log.warn(`Captured API Error: ${errorHandler.capturedError.messages}`);
+
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFoundPage />;
+      }
+
+      errorMessage = errorHandler.renderError();
+    }
+
+    const submitButtonText = loading
+      ? i18n.gettext('Submitting your report…')
+      : i18n.gettext('Submit report');
+
+    const categories = getCategories(i18n);
+    const categoryInputs = {};
+    // eslint-disable-next-line guard-for-in
+    for (const categoryType in categories) {
+      categoryInputs[categoryType] = [];
+      categories[categoryType].forEach((category) => {
+        categoryInputs[categoryType].push(
+          <li className="Feedback-category--li" id={`li-${category.value}`}>
+            <input
+              type="radio"
+              className="Feedback-catgeory"
+              id={`feedbackCategory${category.value}`}
+              name="category"
+              onChange={this.onFieldChange}
+              value={category.value}
+              selected={this.state.category === category.value}
+            />
+            <label
+              className="Feedback--label"
+              htmlFor={`feedbackCategory${category.value}`}
+            >
+              {category.label}
+            </label>
+            <p className="Feedback--help">{category.help}</p>
+          </li>,
+        );
+      });
+    }
+
+    const abuseSubmitted = abuseReport && abuseReport.message !== undefined;
+
+    return (
+      <Page>
+        <div className="Feedback">
+          <Helmet>
+            <title>
+              {i18n.sprintf(
+                i18n.gettext('Submit Feedback for %(type)s %(identifer)s'),
+                {
+                  type: 'Addon',
+                  identifer: addonIdentifier,
+                },
+              )}
+            </title>
+          </Helmet>
+
+          {abuseSubmitted && (
```

Comment:
Ideally this would go in a `renderReportSentConfirmation()` method, and then we could have something like:

```
{abuseSubmitted ? this.renderReportSentConfirmation() : (
  ... the form ...
)}